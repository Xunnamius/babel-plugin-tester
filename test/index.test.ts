/* eslint jest/require-hook: ["error", { "allowedFunctionCalls": ["pluginTesterAsDefault","pluginTester"] }] */

import pluginTesterAsDefault, {
  prettierFormatter,
  unstringSnapshotSerializer,
  pluginTester,
  runPluginUnderTestHere,
  runPresetUnderTestHere
} from '../src/index';

test('prettierFormatter is exported', () => {
  expect.hasAssertions();
  expect(typeof prettierFormatter).toBe('function');
});

test('unstringSnapshotSerializer is exported', () => {
  expect.hasAssertions();
  expect(unstringSnapshotSerializer).toHaveProperty('test');
  expect(unstringSnapshotSerializer).toHaveProperty('print');
});

test('the default export and the named export point to the same function', () => {
  expect.hasAssertions();
  expect(pluginTester).toBe(pluginTesterAsDefault);
});

test('runPluginUnderTestHere is exported', () => {
  expect.hasAssertions();
  expect(typeof runPluginUnderTestHere).toBe('symbol');
});

test('runPresetUnderTestHere is exported', () => {
  expect.hasAssertions();
  expect(typeof runPresetUnderTestHere).toBe('symbol');
});

const uglyTest = `var output = paragraph + "one"+\n\n'paragraph' + \`\\two\``;

pluginTesterAsDefault({
  pluginName: 'captains-log',
  plugin: () => ({ name: 'captains-log', visitor: {} }),
  tests: [
    // ? Test that prettierFormatter is the default `formatResult` function
    { code: uglyTest, output: prettierFormatter(uglyTest) },
    // ? Test that unstringSnapshotSerializer is the active snapshot serializer
    // ! Requires manual inspection of snapshot
    { code: uglyTest, snapshot: true }
  ]
});
