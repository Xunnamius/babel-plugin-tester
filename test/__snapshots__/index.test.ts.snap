// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`captains-log 2. captains-log: 2. captains-log 1`] = `

var output = paragraph + "one"+

'paragraph' + \`\\two\`

      ↓ ↓ ↓ ↓ ↓ ↓

var output = paragraph + 'one' + 'paragraph' + \`\\two\`;

`;

exports[`unknown plugin 4. using jest snapshots: 4. using jest snapshots 1`] = `


function sayHi(person) {
  return 'Hello ' + person + '!'
}
      

      ↓ ↓ ↓ ↓ ↓ ↓

function iHyas(nosrep) {
  return 'Hello ' + nosrep + '!';
}

`;
